models/demos/bert_tiny/demo/demo.py

```diff
@@ -0,0 +1,305 @@
+    hugging_face_reference_model = BertForQuestionAnswering.from_pretrained(model, torchscript=False)
+    pytorch_model = hugging_face_reference_model.eval()
+
+    tokenizer_name = str(model_location_generator(model_name, model_subdir="Bert"))
+    tokenizer = BertTokenizer.from_pretrained(tokenizer_name)
+    config = hugging_face_reference_model.config
+    nlp = pipeline("question-answering", model=hugging_face_reference_model, tokenizer=tokenizer)
+
+    profiler.start(f"preprocessing_parameter")
+    parameters = preprocess_model_parameters(
+        initialize_model=lambda: pytorch_model,
+        device=device,
+        convert_to_ttnn=lambda *_: True,
+    )
+    profiler.end(f"preprocessing_parameter")
+
+    context, question = load_inputs(input_path, batch_size)
+
+    preprocess_params, _, postprocess_params = nlp._sanitize_parameters()
+    preprocess_params["max_seq_len"] = sequence_size
+    inputs = nlp._args_parser({"context": context, "question": question})
+    preprocessed_inputs = []
+    for i in range(batch_size):
+        model_input = next(nlp.preprocess(inputs[0][i], **preprocess_params))
+        single_input = {
+            "example": model_input["example"],
+            "inputs": model_input,
+        }
+        preprocessed_inputs.append(single_input)
+
+    bert_input = tokenizer.batch_encode_plus(
+        zip(question, context),
+        max_length=sequence_size,
+        padding="max_length",
+        truncation=True,
+        return_attention_mask=True,
+        return_token_type_ids=True,
+        return_tensors="pt",
+    )
+
+    position_ids = positional_ids(config, bert_input.input_ids)
+    profiler.start(f"preprocessing_input")
+    ttnn_bert_inputs = preprocess_inputs(
+        bert_input["input_ids"],
+        bert_input["token_type_ids"],
+        position_ids,
+        bert_input["attention_mask"],
+        device=device,
+    )
+    profiler.end(f"preprocessing_input")
+
+    profiler.start(f"inference_time")
+    ttnn_output = bert_for_question_answering(
+        config,
+        *ttnn_bert_inputs,
+        parameters=parameters,
+        device=device,
+    )
+    profiler.end(f"inference_time")
+
+    ttnn_output = (
+        ttnn.to_torch(ttnn.from_device(ttnn_output)).reshape(batch_size, 1, sequence_size, -1).to(torch.float32)
+    )
+
+    ttnn_start_logits = ttnn_output[..., :, 0].squeeze(1)
+    ttnn_end_logits = ttnn_output[..., :, 1].squeeze(1)
+
+    model_answers = {}
+    profiler.start("post_processing_output_to_string")
+    for i in range(batch_size):
+        tt_res = {
+            "start": ttnn_start_logits[i],
+            "end": ttnn_end_logits[i],
+            "example": preprocessed_inputs[i]["example"],
+            **preprocessed_inputs[i]["inputs"],
+        }
+
+        tt_answer = nlp.postprocess([tt_res], **postprocess_params)
+
+        logger.info(f"answer: {tt_answer['answer']}\n")
+        model_answers[i] = tt_answer["answer"]
```

Comment:
test should fail if answers are wrong or perf slower than expected

Reply:
Bert-Tiny model results are not promising but got approval from @mbahnasTT for the demo results.

Reply:
Pytorch results does not align with the dataset for most of the cases.
However, Pytorch and TTNN results match in most cases, and the results can be found here: [Link](https://docs.google.com/spreadsheets/d/1dBKFjR5tvsdn3oCvyMyOX1D_z_LwMWEI-q7MSQjvo_o/edit?gid=1101027898#gid=1101027898)

Accuracy between TTNN model and SQUADv2 reference:
```
Eval: exact: 14.423076923076923 
F1: 22.086671341479036
```

Accuracy between PyTorch model and SQUADv2 reference:
```
exact: 19.23076923076923 
F1: 24.28325947556717
```

Accuracy between TTNN and PyTorch model:
```
exact: 58.65384615384615
F1: 0.5875457875457875
```